File: CS50_2017/Lecture1/Problemset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int getCoin(int n);

int main(void)
{
    int counter = 0;
    float change = -1;
    int cents = 0;
    while (change < 0)
    {
        printf("How much is the change?\n");
        change = get_float();

        cents = roundf(change * 100);
    }
    while (cents != 0)
    {
        cents -= getCoin(cents);
        counter++;

    }
    printf("%i\n", counter);
}

int getCoin(int n)
{
    int result=0;
    if (n >= 25) {
        result = 25;
    } else if (n >= 10) {
        result = 10;
    } else if (n >= 5) {
        result = 5;
    } else if (n >= 1) {
        result = 1;
    }
    return result;
}

File: CS50_2017/Lecture8/Problemset6/greedy.py
import sys
import cs50

def main():
    counter = 0;
    change = -1;
    cents = 0;
    while change < 0:
        print("How much is the change?")
        change = cs50.get_float()

        cents = round(change * 100)
    while cents != 0:
        cents -= getCoin(cents)
        counter += 1
    print(counter)

def getCoin(n):
    result = 0
    if n >= 25:
        result = 25
    elif n >= 10:
        result = 10
    elif n >= 5:
        result = 5
    elif n >= 1:
        result = 1
    return result

if __name__ == "__main__":
    main()
    